api/pkg/metadata/service.go

```diff
@@ -51,20 +51,55 @@ func (s *APIService) SyncMetadata(request *restful.Request, response *restful.Re
 }
 
 func (s *APIService) ListMetadata(request *restful.Request, response *restful.Response) {
-	log.Infof("Received request for Listmetadata details: %s\n", request.PathParameter("id"))
-
-	//id := request.PathParameter("id")
+	log.Infof("received request for list metadata details: %s\n", request.PathParameter("id"))
 
 	ctx := common.InitCtxWithAuthInfo(request)
 
-	res, err := s.metaClient.ListMetadata(ctx, &mt.ListMetadataRequest{Limit: 1000})
+	listMetadataRequest, err := GetListMetaDataRequest(request)
+
+	if err != nil {
+		log.Errorf("Failed to construct list metadata request err: \n", err)
+		response.WriteError(http.StatusBadRequest, err)
+		return
+	}
+
+	//* calling  the ListMetaData method from metadata manager m8s
+	res, err := s.metaClient.ListMetadata(ctx, &listMetadataRequest)
 	log.Info("Get metadata details res.......:.", res)
 	if err != nil {
 		log.Errorf("Failed to get metadata details err: \n", err)
 		response.WriteError(http.StatusNotFound, err)
 		return
 	}
 
-	log.Info("Get metadata details successfully.")
+	log.Info("got metadata details successfully.")
 	response.WriteEntity(res.Buckets)
 }
+
+//* This function fetches the request parameters from the request and assigns them default values if not present.
+//* It returns ListMetadataRequest for ListMetaData API call
+func GetListMetaDataRequest(request *restful.Request) (listMetadataRequest mt.ListMetadataRequest, err error) {
+	typeOfCloudVendor := request.PathParameter("type")
+	backendName := request.PathParameter("backendName")
+	bucketName := request.PathParameter("bucketName")
+	objectName := request.PathParameter("objectName")
+	sizeOfObjectInBytes, err := common.GetSizeRequestParamAsInt64(request, "sizeOfObject")
+
+	if err != nil {
+		return mt.ListMetadataRequest{}, err
```

Comment:
Please avoid returing err messge directly to user. You may return.

type of sizeOfObject is invalid. It should be Integer type.

Reply:
Done